### PR TITLE
Require Ruby 3 and Rails 6

### PR DESCRIPTION
**Pull request by jeffbax**
And clear spec deprecation warnings for Rails 6.1

### CODE REVIEW

**Review by bsaris-ut — APPROVED** [review]
This seems reasonable, thanks for taking care of this.

File: Gemfile.rails_version [hunk 1]
```diff
@@ -2,22 +2,19 @@
...
 when "master"
   {github: "rails/rails"}
-when /^5.2/, /^5.1/, /^5.0/
+when /^6.1/, /^6.0/
   gem 'bootstrap-sass', '~> 3.3'
   gem 'jquery-rails'
   gem "sprockets-rails", "~> 3.2"
-
-  # A JS runtime is required for Rails 4.0+
-  gem 'therubyracer', '~> 0.12.3', platforms: :ruby
```
Comment (jeffbax) [inline Gemfile.rails_version:20]: this didn't seem like it was useful for the spec suite which is the only thing I really run so I just ripped it out.

it barely builds on intel macs and not arm macs at all

File: Gemfile.rails_version [hunk 2]
```diff
@@ -2,22 +2,19 @@
...
-ruby '~> 2.6.7'
+ruby '~> 3.0.0'
 
 gem 'rails', "~> #{env_rails_version}"
 
 case env_rails_version
 when "master"
   {github: "rails/rails"}
-when /^5.2/, /^5.1/, /^5.0/
+when /^6.1/, /^6.0/
```
Comment (jeffbax) [inline Gemfile.rails_version:14]: why do less?

File: Rakefile [hunk 3]
```diff
@@ -27,7 +27,7 @@ namespace :testbed do
       f.write("application \"config.time_zone='Rome'\"\n")
       f.flush
       # don't run bundle install until the Gemfile modifications
-      sh "bundle exec rails new testbed --skip-turbolinks --skip-bundle -m #{f.path}"
+      sh "bundle exec rails new testbed --skip-turbolinks --skip-bundle --skip-webpack-install -m #{f.path}"
```
Comment (jeffbax) [inline Rakefile:30]: another testbed setup error that didn't seem to have a point

File: lib/custom_model_naming.rb [hunk 4]
```diff
@@ -11,7 +11,7 @@ module CustomModelNaming
   module ClassMethods
     def model_name
       @model_name ||= begin
-        namespace = parents.detect do |n|
+        namespace = module_parents.detect do |n|
```
Comment (jeffbax) [inline lib/custom_model_naming.rb:14]: everything after this was a dep warning

File: .travis.yml [hunk 5]
```diff
@@ -1,17 +0,0 @@
-language: ruby
```
Comment (jeffbax) [inline .travis.yml:1]: this seemed less than useful to stick around given we don't run the main repo CI

File: .gitignore [hunk 6]
```diff
@@ -29,3 +29,4 @@ testbed
 surveys/fixtures/*.yml
 .rvmrc
 .sass-cache/**/*.sassc
+/.bundle
```
Comment (jeffbax) [inline .gitignore]:
```suggestion
```

File: Gemfile.rails_version (hunk 1 above)
Comment (bsaris-ut) [reply Gemfile.rails_version:20]: I'm always down to remove things from this gem and ultimately want to remove the gem all together.

**Pull request merged**